### ISSUE TITLE
Merging summary(), anova() results not working as expected, esp with categorical variables

### ISSUE THREAD

**Issue by jenniferthompson** [jenniferthompson/JTHelpers#16]
When no `labeldf` is passed, unexpected results for categorical variables (this is from `fit.mult.impute()` with `fitter = lrm`, not sure if it affects others):

```
12                    race.comb                                      0.3    1
13  race.comb...Non.white.White   1.00  2.00 0.83 (0.43, 1.62)           <NA>
```

**Comment by jenniferthompson** [comment]
Oh dear, all the nonlinear terms are messed up too.

**Comment by jenniferthompson** [comment]
Passing a `labeldf` helps with nonlinear terms, but not with categorical variables.

**Comment by jenniferthompson** [comment]
Is `rmRows` not working properly? Should be removing extra NL + interaction rows by default.

**Comment by jenniferthompson** [comment]
This seems to be due to an R version change!

Desktop at work: 3.4.4, have noticed no issues
Laptop: currently 3.5

Previously, `as.data.frame(summary([lrm() model fit]))` wouldn't print, because there can be duplicate rownames (ie, many rows titled ` Odds Ratio`). New behavior seems to be to [automatically differentiate rownames](https://cran.r-project.org/bin/windows/base/NEWS.R-devel.html). While I appreciate the new behavior, the change creates temporary madness in my code.

```
hosp.type...Veteran.Civilian
X.Odds.Ratio.13             
```

Bottom line, relying on rownames is the devil. 👿

Solution TBD. May do a quick fix for today and then a more reliable one.

**Comment by jenniferthompson** [comment]
Quick fix for categorical variables:

```
  ## -- summary.rms() results --------------------------------------------------------------------
  ## Original: often wouldn't print due to duplicate rownames; with R v 3.5,
  ## duplicate behavior changed. Get rownames **before** converting to
  ## data.frame to preserve info without extra characters.
  mod.rows <- rownames(summary(rmsObj))
  mod.sum <- as.data.frame(summary(rmsObj))
  mod.sum$quantity <- mod.rows
  rownames(mod.sum) <- NULL
```

And:

```
  ## -- anova.rms() results ----------------------------------------------------------------------
  ## Create data frame of anova() results, add column to indicate variable/quantity
  ## See above about R 3.5 behavior change re rownames
  mod.anova.rows <- rownames(anova(rmsObj))
  mod.anova <- as.data.frame(anova(rmsObj))
  mod.anova$line <- mod.anova.rows
  rownames(mod.anova) <- NULL
  mod.anova$var <- gsub('^ ', '', gsub(' +\\(.*\\)$| : .*$', '', mod.anova$line))
  ```
